pkg/http/listener.go

```diff
@@ -88,6 +89,19 @@ type httpListener struct {
 	errorLogFunc           func(error, string, ...interface{}) // function to be called on errors.
 }
 
+// isExpectedErr returns true if error is due to a network timeout or
+// io.EOF and false otherwise
+func isExpectedErr(err error) bool {
```

Comment:
@balamurugana  these unit tests additionally document the errors we avoid logging via test cases. So, I intend to retain it as a separate function with the unit tests.